File: lib/screen/TorusScreen.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Point_1 = require("../Point");
const Screen_1 = require("../Screen");
const torusMod = (n, size) => n < 0 ? size - Math.abs(n) % size : n % size;
class TorusScreen extends Screen_1.default {
    constructor(width, height, shapes = []) {
        super(width, height, (point, screen_width, screen_height) => new Point_1.default(torusMod(point.x, screen_width), torusMod(point.y, screen_height)), shapes);
    }
    toString() {
        return this.occupiedArea().map((row) => row.join('')).join('\n');
    }
}
exports.default = TorusScreen;

File: src/screen/TorusScreen.py
from src.Point import Point
from src.Screen import Screen
from src.Shape import Shape

def torusMod(n, size):
  if n < 0:
    return (size - abs(n)) % size
  else:
    return n % size

def torusMap(point, screen_width, screen_height, *_):
  return Point(
    torusMod(point.x, screen_width),
    torusMod(point.y, screen_height)
  )

class TorusScreen(Screen):
  def __init__(self, width, height, shapes = None):
    super().__init__(width, height, torusMap, shapes)